snuba/clickhouse/astquery.py

```diff
@@ -91,7 +92,14 @@ def _sql_data_list(self) -> Sequence[Tuple[str, str]]:
 
         where_clause = ""
         if self.__condition:
-            formatted_condition = self.__condition.accept(formatter)
+            if state.get_config("infix_where_format", 0):
+                conditions = get_first_level_conditions(self.__condition)
+                formatted_condition = " AND ".join(
+                    c.accept(formatter) for c in conditions
+                )
+            else:
+                formatted_condition = self.__condition.accept(formatter)
+
```

Comment:
This seems like it logically belongs in the ClickHouse expression formatter. I guess it would be applied elsewhere too if so (like prewhere and having). But is there any reason we don't want that to be the case and  just apply the special rule here?

Reply:
It is sensibly more complicated there, because we would have to deal with operators precedence in a place where the visitor does not have context on the expression.
Here I intentionally dealt only with the first level since this is just to make the query a bit more readable inn the querylog.

In order to preserve the correctness there are two ways, either I add parentheses around every binary boolean expression doing something like
`(((A AND B) AND (C OR X)) AND D)`
or, in order to obtain an expression like
`A AND B AND (C OR X) AND D`
I would need to pre-flatten the and/or function or propagate the context within the visitor. So having a main method for and/or and a secondary method for their children.

Not sure it is worth, but they can certainly be achieved. What do you think ?

Reply:
moved to the visitor and supported both AND and OR